assertion/function/analyzer.go

```diff
@@ -223,15 +225,29 @@ func run(pass *analysis.Pass) (result interface{}, _ error) {
 	var errs []error
 	funcTriggers := make([][]annotation.FullTrigger, funcIndex)
 	triggerCount := 0
+	funcResults := map[*types.Func]*functionResult{}
 	for r := range funcChan {
 		if r.err != nil {
 			errs = append(errs, r.err)
 		} else {
 			funcTriggers[r.index] = r.triggers
 			triggerCount += len(r.triggers)
+
+			funcObj, ok := pass.TypesInfo.ObjectOf(r.funcDecl.Name).(*types.Func)
+			if !ok {
+				continue
+			}
+			funcRes := r
```

Comment:
Why the intermediate var here and not just: `funcResults[funcObj] = &r`?

Reply:
Good question. Thanks to the special design choice of Go, `r` in each iteration is re-updated with a new value, which means the address of `r` never changes. This is to make sure we get the correct address in every iteration; otherwise if we do `&r`, we always have the same address, which does not change with the loop.

Reply:
> Thanks to the special design choice of Go

A phrase that crops up all to often on NilAway development 😅 But, right, makes sense.